chain/actors/builtin/market/market.go

```diff
@@ -140,3 +141,15 @@ func EmptyDealState() *DealState {
 		LastUpdatedEpoch: -1,
 	}
 }
+
+// returns the earned fees and pending fees for a given deal
+func (deal DealProposal) GetDealFees(height abi.ChainEpoch) (abi.TokenAmount, abi.TokenAmount) {
+	tf := big.Mul(deal.StoragePricePerEpoch, big.NewInt(int64(deal.EndEpoch-deal.StartEpoch)))
+
+	ef := big.Mul(deal.StoragePricePerEpoch, big.NewInt(int64(height-deal.StartEpoch)))
```

Comment:
Should also be capped at total fees / deal expiration

Reply:
Oops, that was silly...thanks!